File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        if (s == ""):
            return 0

        res, l = 0, 0
        hashmap = {s[0] : 1}

        for r in range(1, len(s)):
            hashmap[s[r]] = hashmap.get(s[r], 0) + 1
            cur = r - l + 1

            if cur - max(hashmap.values()) <= k:
                res = max(res, cur)
            else: 
                hashmap[s[l]] -= 1
                l += 1
        
        return max(res, r - l)

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap <Integer, Integer> complement = new HashMap<>();
        int [] res = new int [2];

        for (int i = 0; i < nums.length; i++) {
            int currInt = nums[i];
            if (complement.containsKey(target - currInt)) {
                res[0] = complement.get(target - currInt);
                res[1] = i;
                break;
            }
            complement.put(currInt, i);
        }
        return res;
    }
}
